TelemetryEvents.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+private let PrefKeySearches = "Telemetry.Searches"
+
+public class SearchTelemetryFactory {
+    // For data consistency, the strings used here are identical to the ones reported in Android.
+    public enum Source: String {
+        case URLBar = "actionbar"
+        case QuickSearch = "listitem"
+        case Suggestion = "suggestion"
+    }
+
+    private init() {}
+
+    public class func makeEvent(engineID engineID: String, source: Source) -> TelemetryEvent {
+        return SearchTelemetryEvent(engineWithSource: "\(engineID).\(source.rawValue)")
```

Comment:
Aside from not requiring telemetry clients to have a `prefs` handle, another reason for this indirection is the fact that `Telemetry.swift` and `TelemetryEvents.swift` are in two different build targets. `Telemetry.swift` is shared (i.e., it can be accessed from the main app or any of our share extensions). `TelemetryEvents.swift` is specific to our main client app, meaning it can use any classes that are only in that target.

I had actually intended for the `makeEvent` signature to look like this:
```
makeEvent(engine engine: OpenSearchEngine, source: Source) -> TelemetryEvent
```

That is, it accepts an actual typed engine instead of a plain string, which makes it ever-so-slightly less error-prone. This wouldn't be possible if `TelemetryEvents.swift` was also in the shared target since `OpenSearchEngine` is client-specific.

I'll update the PR with this change -- thanks for the reminder!